modules/virt-creating-custom-resources-hpp.adoc

```diff
@@ -0,0 +1,72 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virtual_disks/virt-configuring-local-storage-for-vms.adoc
+
+:_content-type: PROCEDURE
+[id="virt-creating-custom-resources-hpp_{context}"]
+= Creating host path provisioner (HPP) custom resources
+
+If you upgrade to {VirtProductName} version 4.10 on an existing cluster, you are not required to change your HPP custom resource. You continue using your existing storage classes. You do not have to modify the HPP custom resource for the legacy provisioner DaemonSet.
+
+== Creating the legacy HPP custom resource
+
+If you are installing {VirtProductName} version 4.10 on a new cluster, you must create an HPP legacy custom resource:
+
+[source,yaml]
+----
+apiVersion: hostpathprovisioner.kubevirt.io/v1beta1
+kind: HostPathProvisioner
+metadata:
+  name: hostpath-provisioner
+spec:
+  imagePullPolicy: IfNotPresent
+  pathConfig:
+    path: "/var/hpvolumes"
+    useNamingPrefix: false
+ workload:
+   nodeSelector:
+     kubernetes.io/os: linux
+----
+
+== Creating the CSI driver provisioner custom resource
+
+The legacy provisioner DaemonSet and the CSI driver DaemonSet are supported in parallel for a number of releases. However, at some point, the legacy provisioner DaemonSet will no longer be supported. To prepare for this change,
+create a CSI driver custom resource using a basic storage pool.
+
+Storage pools allow you to specify the source and path that are used by the CSI driver. They also allow you to name the storage pool that is used in the storage class.
+
+.Procedure
+
+. Create a YAML file for the CSI driver provisioner custom resource. For example:
++
+[source,terminal]
+----
+$ touch hostpathprovisioner_cr.yaml
+----
+
+. Edit the file. For example:
++
+[source,yaml]
+----
+apiVersion: hostpathprovisioner.kubevirt.io/v1beta1
+kind: HostPathProvisioner
+metadata:
+ name: hostpath-provisioner
+spec:
+ imagePullPolicy: IfNotPresent
+ storagePools: <1>
+   - name: local <2>
+     path: "/var/hpvolumes"
```

Comment:
Yeah, we should mention that the path should be some place that is read/write (on RHCOS many directories are read only). The ones we normally recommend are either `/var/something` or `/mnt/something` since both /var and /mnt are read/write in RHCOS.

Reply:
Thanks. I updated a few of the others callouts as well to:

[source,yaml]
----
apiVersion: hostpathprovisioner.kubevirt.io/v1beta1
kind: HostPathProvisioner
metadata:
 name: hostpath-provisioner
spec:
 imagePullPolicy: IfNotPresent
 storagePools: <1>
   - name: local <2>
     path: "/var/myvolumes" <3>
 workload:
   nodeSelector:
     kubernetes.io/os: linux
----
<1> The `storagePools` specification is an array to which you can add multiple entries.
<2> In the CSI driver custom resource, there is a single storage pool named “local” that uses the path `/var/hpvolumes`.The storage pool assumes you have created a directory (`hpvolumes`) located at the node-level for creation of volumes.
<3> Read/write access required.